ooniprobe/Model/Database/Result.mm

```diff
@@ -108,6 +108,8 @@ -(Measurement*)getLastMeasurement{
 -(float)getRuntime{
     Measurement *first = [self getFirstMeasurement];
     Measurement *last = [self getLastMeasurement];
+    if (first == nil || last == nil)
+        return 0;
```

Comment:
I don't think it's a good idea to return a valid runtime value. I think it should rather be treated as an error and logged as such.
My suggestion is to throw an exception which mentions explicitly what is going on, see: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Exceptions/Tasks/HandlingExceptions.html.

Reply:
Even if I throw an exception I still need to return a value. We can't make the app crash

Reply:
I think that we need to record this somehow as an unexpected event that needs to be investigated as it's probably hiding some bugs in the app. Making the app crash (which is what would happen in the case of a NULL pointer exception) with a specific exception is better than hiding a bug.